### DISCUSSION TITLE
Remix V2 on Vite: Do we lose out on any vite features by not starting our custom server with `vite dev`?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [remix-run/remix#7995] [Q&A] [answered]
## Question
Do we lose out on any `vite` features by not starting our custom server with `vite dev`?

## Background Context
I noticed in the Remix Vite upgrade docs that you could start your Remix Dev server using vite like so:

**package.json**
```
"dev": "vite dev"
````` 

Or you could continue starting your dev server like this (assuming you are using custom server, like express template which exposes `server.js`)
```
"dev": "node server.js"
```

Remix App Server you can use `vite dev`:
https://remix.run/docs/en/main/future/vite#migrating-from-remix-app-server
![image](https://github.com/remix-run/remix/assets/6743796/153e7c23-2c4d-44f9-ab74-5de3a8e1fe76)

But they recommend if you're using remix with a custom server (ex: express template which exposes server.js) to call server directly:

https://remix.run/docs/en/main/future/vite#migrating-from-a-custom-server
![image](https://github.com/remix-run/remix/assets/6743796/7e14913c-320e-42be-bd61-e6267a04ef08)

## References
- Official Remix Discord #Help Thread Link: https://discord.com/channels/770287896669978684/1173713912064118785/1173713912064118785

**Comment by pcattori** [comment, marked as answer]
>  Do we lose out on any vite features by not starting our custom server with `vite dev`?

Nope. If you are delegating assets and Remix request handler stuff to Vite like shown in the docs, then you are only limited by whatever you choose as your server (e.g. Express). Note that in this case, any `vite.config.ts` fields that only affect the Vite server won't be relevant since you aren't running the Vite server, but any features provided by those options should be implementable in your own custom server too.
